### PR TITLE
[WIP] Sharded atomics

### PR DESCRIPTION
**Pull request by efirs**
On write increment random shard of N.
On read sum up the value from all the shards.

This will be used by statistics calculation, such as data size.

### CODE REVIEW

File: store/kv/atomic.go [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+	n := rand.Intn(s.numShards) //nolint:gosec
+
+	key = append(key, AtomicSuffixSeparator)
+	key = append(key, n)
+
+	return s.kv.AtomicAdd(ctx, table, key, inc)
+}
+
+// AtomicRead calculates the value by summing all the shards.
+func (s *shardedAtomics) AtomicRead(ctx context.Context, table []byte, key Key) (int64, error) {
```
Comment (himank) [inline store/kv/atomic.go:55]: When we will be diffing and updating that can be done on any random shard, so we need one more API to randomly read from any shard.

File: store/kv/atomic.go [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+	"context"
+	"math/rand"
+)
+
+const (
+	DefaultNumShards      = 1024
+	AtomicSuffixSeparator = "__A__"
+)
+
+type ShardedAtomics interface {
```
Comment (himank) [inline store/kv/atomic.go:27]: As we are introducing this interface, should we move `AtomicAdd`/`AtomicRead` from the kv interface to a `NonShardedAtomics` interface? Or remove them.

File: store/kv/atomic.go (hunk 1 above)
Comment (efirs) [reply store/kv/atomic.go:55]: Individual shard value is no better then picking a random number,
I don't see a value accessing it individually.

File: store/kv/atomic.go (hunk 2 above)
Comment (efirs) [reply store/kv/atomic.go:27]: Not a bad idea. I think how to do it properly.

File: store/kv/atomic.go (hunk 1 above)
Comment (himank) [reply store/kv/atomic.go:55]: Never mind, I was still thinking that we are not using atomics, and then we need to read in a transaction. We don't need the API to read from the random shard, this is good.

**Pull request closed without merging**